test/e2e/node_feature_discovery_test.go

```diff
@@ -725,24 +804,26 @@ func waitForNfdNodeAnnotations(cli clientset.Interface, expected map[string]stri
 
 type k8sLabels map[string]string
 
-// waitForNfdNodeLabels waits for node to be labeled as expected.
-func waitForNfdNodeLabels(cli clientset.Interface, expected map[string]k8sLabels) error {
+// checkForNfdNodeLabels waits and checks that node is labeled as expected.
+func checkForNodeLabels(cli clientset.Interface, expectedNewLabels map[string]k8sLabels, oldLabels map[string]string) error {
```

Comment:
Hmm, this doesn't work as the `oldLabels` might be different between nodes. Similar to `expectedNewLabels` you should have per-node old-labels like `oldLabels map[string]k8sLabels`.

Actually, probably the easiest would be to have just `oldNodes []corev1.Node` so you could just give the full nodelist as an arg when calling `checkForNodeLabels`, with no need to construct anything like `map[string]k8sLabels`...

Reply:
@marquiz  I see that constructing the `map[string]k8sLabels` would be better because inside the `checkForNodeLabels` function, we can access the labels of a specific node in `O(1)` time complexity, but without the construction, we need to access the labels of a node in `O(n)` time complexity where `n is the number of nodes. please correct me if I'm wrong.

Reply:
Basically true, but it is insignificant, especially in this case of e2e tests. I think less lines-of-code and better readability is preferable.